src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -688,7 +718,8 @@ void StatefulWriter::send_changes_separatedly(
                 auto unsent_change_process =
                         [&](const SequenceNumber_t& seqNum, const ChangeForReader_t* unsentChange)
                         {
-                            if (unsentChange != nullptr && unsentChange->isRelevant() && unsentChange->isValid())
+                            if (unsentChange != nullptr && remoteReader->rtps_is_relevant(unsentChange->getChange()) &&
```

Comment:
As per our conversation, do not use `rtps_is_relevant()` and keep `isRelevant()`. The internal state of the `ChangeForReader_t` already has the updated relevance value given by `rtps_is_relevant()`

Also applicable to the rest of the file where the check has been substituted.